version.py

```diff
@@ -40,18 +40,28 @@
 
 
 def call_git_describe(prefix='', cwd='.'):
-    cmd = 'git describe --tags --match %s[0-9]*' % prefix
+    version_cmd = 'git describe --tags --match %s[0-9]*' % prefix
+    last_tag_cmd = 'git describe --tags --match %s[0-9]* --abbrev=0' % prefix
     try:
         version = subprocess.run(
-            cmd.split(),
+            version_cmd.split(),
             stdout=subprocess.PIPE,
-            check=True,
+            check=False,
         ).stdout.decode()
+
+        last_tag = subprocess.run(
+            last_tag_cmd.split(),
+            stdout=subprocess.PIPE,
+            check=False,
+        ).stdout.decode()
+
+        assert ' ' not in prefix
         version = version.strip()[len(prefix):]
-        if '-' in version:
-            version = '{}+dev{}.{}'.format(*version.split('-'))
+        last_tag = last_tag.strip()[len(prefix):]
+        if version != last_tag:
+            version = f"{last_tag}+dev{version[len(last_tag) + 1:].replace('-', '.')}"
```

Comment:
Please replace `+dev` with `-dev` according to our discussion on Rocket.

Of course do it in a separate (non-fixup) commit since this does significantly change the output so it should be visible as a separate change.

Reply:
Done

Reply:
@cameel see my qustion on Rocket please

Reply:
The consensus seems to be that the fix should be elsewhere. So let's leave it at `+dev`.